### ISSUE TITLE
Use Comments only when Needed.

### ISSUE THREAD

**Issue by Sree-Vandana** [Sree-Vandana/CodeReadingAndReview2021#6]
### Rule Description

- Use Comments when needed to provide `clarification` about some lines of code or to provide some `code segmentation` _(as mentioned in issue #13)_ or mention some `warnings` about the code or `TO DO Items`.

- Remove the comments, when the code it is talking about changes. And remove them altogether during refactoring, when its no longer needed.

- Do not make the comments too wordy. An example of too wordy comment [Info.js, Line 34 to 36](https://github.com/Sree-Vandana/CodeReadingAndReview2021/blob/ee4779bce9bdea9061aee632109ec145c84b6ead/frontend-web-dev-project-main/frontend-project/src/components/Info.js#L34)

- Some of the **Good Examples** of comments that provide clarification are: This file has so many good examples, few lines are [PersonalInfo.js, Lines 126, 130, 140, 146](https://github.com/Sree-Vandana/CodeReadingAndReview2021/blob/ee4779bce9bdea9061aee632109ec145c84b6ead/frontend-web-dev-project-main/frontend-project/src/components/PersonInfo.js#L126) 
- A **bad example** can be, [Info.js, Line 19](https://github.com/Sree-Vandana/CodeReadingAndReview2021/blob/ee4779bce9bdea9061aee632109ec145c84b6ead/frontend-web-dev-project-main/frontend-project/src/components/Info.js#L19) clarification that it stores rating, need not be mentioned, because the variable name is clear enough.

- Try to associate a comment with a keyword to provide more clarification. Some Key Words to use are: 
      1. WARNING
      2. TODO
      3. INFO
      4. ALERT

 **Examples**

    // WARNING: Below function can throw "ValueCannotBeNull Exception"

    // TODO: Waiting for API backend to be deployed. Will Implement soon

    // INFO: we assume the default max is 100 unless it is TMDB, then it is 10
    

### Why it is needed

- Comments help to navigate the code base much easily,  without having to go through the entire code line by line. Just by reading the comments it is easy to understand what some lines of code does.

- Programmers can't realistically maintain comments. During the development process, the code changes a lot, to improve the performance or while refactoring, the code structure changes as well as what a segment of code does, and comments can be easily overlooked. And problems can arise when a dangling comment is left without removing it, when the code it represents got modified or removed. This will cause more problems than it resolves things. So use comments only if they are necessary.

- Associating a comment with a keyword, helps to skim through the comment easily. And just like how alert we stay during a red light or danger sign, seeing the `WARNING `keyword, we can be sure the person reading the code  will read it. It can be hard to follow this in real time, but at least we can use most important key words like, `WARNING` and `ALERT`.
**Example**: In [Info.js, Line 15 to 25](https://github.com/Sree-Vandana/CodeReadingAndReview2021/blob/ee4779bce9bdea9061aee632109ec145c84b6ead/frontend-web-dev-project-main/frontend-project/src/components/Info.js#L15) Its ok to not provide the keyword `INFO` for these comments, and it is better not to provide key word when the comments are so crowded.

**Comment by Sree-Vandana** [comment]
Scanning through the code, to analyze if a comment is needed or not. While scanning I realized, its better to have a key word associate with them, to simply tell which category it belongs to and which comment to focus on. 

Some key words can be `WARNING`, `TODO`, `CLARIFICATION`

**Comment by shashank114** [comment]
Review completed.
This issue can be closed.
